### PR TITLE
HOSTEDCP-1896: [release-4.15] Allow setting Kube APIServer maximum requests in flight

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Adds support for overriding default maximum requests in flight for Kubernetes API server. This will allow service providers to rate limit the Kubernetes API server based on available resources.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1896](https://issues.redhat.com/browse/HOSTEDCP-1896)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kas/params.go [hunk 1]
```diff
@@ -116,6 +124,12 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 		params.Image = hcp.Spec.Configuration.Image
 		params.Scheduler = hcp.Spec.Configuration.Scheduler
 	}
+	if reqInflight := hcp.Annotations[hyperv1.KubeAPIServerMaximumRequestsInFlight]; reqInflight != "" {
+		params.MaxRequestsInflight = reqInflight
```
Comment (jparrill) [inline control-plane-operator/controllers/hostedcontrolplane/kas/params.go:128]: Do we need to validate the param is correct before put the annotation content directly as a value?

**Review by jparrill — COMMENTED** [review]
Small question, otherwise lgtm.

File: control-plane-operator/controllers/hostedcontrolplane/kas/params.go [hunk 2]
```diff
@@ -116,6 +124,12 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 		params.Image = hcp.Spec.Configuration.Image
 		params.Scheduler = hcp.Spec.Configuration.Scheduler
 	}
+	if reqInflight := hcp.Annotations[hyperv1.KubeAPIServerMaximumRequestsInFlight]; reqInflight != "" {
```
Comment (jparrill) [inline control-plane-operator/controllers/hostedcontrolplane/kas/params.go:127]: Do we need to validate the param is correct before put the annotation content directly as a value?

Comment (sjenning) [reply control-plane-operator/controllers/hostedcontrolplane/kas/params.go:127]: This is a backport of https://github.com/openshift/hypershift/pull/4551

**Review by sjenning — APPROVED** [review]

**Pull request merged**